prow/cmd/deck/main.go

```diff
@@ -1325,22 +1325,24 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface, log *logrus.Entry) htt
 }
 
 // canTriggerJob determines whether the given user can trigger any job.
-func canTriggerJob(user string, pj prowapi.ProwJob, cfg *prowapi.RerunAuthConfig, cli prowgithub.RerunClient, pluginAgent *plugins.ConfigAgent, log *logrus.Entry) (bool, error) {
-	auth, err := cfg.IsAuthorized(user, cli)
-	if auth {
-		return true, nil
-	}
-	if err != nil {
-		return false, err
-	}
-	jobPermissions := pj.Spec.RerunAuthConfig
-	auth, err = jobPermissions.IsAuthorized(user, cli)
-	if auth {
-		return true, nil
+func canTriggerJob(user string, pj prowapi.ProwJob, cfg prowapi.RerunAuthConfig, cli prowgithub.RerunClient, pluginAgent *plugins.ConfigAgent, log *logrus.Entry) (bool, error) {
```

Comment:
Previously the PJ rerunAuth, the config rerunAuth, or the `/test` logic was respected even if others were present. Now if a PJ rerunAuth is present the other two checks are skipped which is a breaking change.

Reply:
The only behavior I changed was to check job-level before checking config-level rerun, to be consistent with the precedence of highest specificity. Should I swap this back?

Reply:
Ah sorry, I misread the combined diff. If I'm reading this correctly now this should behave identically to the previous ordering since it is just a logical disjunction which must be commutative? Wouldn't the only reason to change this be if one of the checks was more expensive and would be better to try after the cheaper checks?

Reply:
Oh, yes good point; you are right. Due to a common boolean return and the disjunction of the statements this is commutative and thus equivalent. Regarding computational cost, based on current usage it is difficult to say which is more expensive; they call the same method. In the case where the Prow config has an `allow_anyone` then it is much faster (no http calls) to check that first, so I will revert ordering to what it was previously.

Reply:
Reverted.